src/api/db/data/20250131094818_nullify_targets_on_bs_request_actions.rb

```diff
@@ -3,30 +3,12 @@
 class NullifyTargetsOnBsRequestActions < ActiveRecord::Migration[7.0]
   # rubocop:disable Rails/SkipsModelValidations
   def up
-    bs_request_actions = BsRequestAction.where('target_project_id IS NOT NULL AND target_package_id IS NULL')
-    bs_request_actions.in_batches do |batch|
-      batch.find_each do |action|
-        target_project = Project.find_by(name: action.target_project)
-        if target_project.nil?
-          action.update_columns(target_project_id: nil)
-        end
-      end
+    BsRequestAction.joins('LEFT JOIN projects ON bs_request_actions.target_project_id = projects.id WHERE projects.id IS NULL').in_batches do |batch|
+      batch.update_all(target_project_id: nil)
     end
 
-    bs_request_actions = BsRequestAction.where('target_project_id IS NOT NULL AND target_package_id IS NOT NULL')
-    bs_request_actions.in_batches do |batch|
-      batch.find_each do |action|
-        target_project = Project.find_by(name: action.target_project)
-        if target_project.nil?
-          action.update_columns(target_project_id: nil, target_package_id: nil)
-          next
-        end
-
-        target_package = Package.find_by_project_and_name(action.target_project, action.target_package)
-        if target_package.nil?
-          action.update_columns(target_package_id: nil)
-        end
-      end
+    BsRequestAction.joins('LEFT JOIN packages ON bs_request_actions.target_package_id = packages.id WHERE packages.id IS NULL').in_batches do |batch|
+      batch.update_all(target_package_id: nil)
```

Comment:
Aren't all of the `target_package_id` fields of those records of that query `nil` already?